spec/unit/functions/call_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'puppet'
+require 'spec_helper'
+require 'puppet_spec/compiler'
+
+require 'matchers/resource'
+
+describe 'the call method' do
+  include PuppetSpec::Compiler
+  include Matchers::Resource
+
+  context "should be callable as" do
+    it 'call on a built-in 4x Ruby API function' do
+      expect(compile_to_catalog(<<-CODE)).to have_resource('Notify[a]')
+          $a = call('split', 'a-b-c', '-')
+          notify { $a[0]: }
+        CODE
+    end
+
+    it 'call on a Puppet language function with no arguments' do
+      expect(compile_to_catalog(<<-CODE)).to have_resource('Notify[called]')
+          function call_me() { 'called' }
```

Comment:
That was the bug I was referring to on Slack that I thought I had found. I've tested the call function in a full environment and it works for namedspaced Puppet language functions. However, it does not work for a non-namespaced Puppet language function. And that seems like a bug in internal_call_function to me. Based on my reading of the code and the API notes it sounds like internal_call_function should work equally for all types of functions. The loader seems unable to find it, but how/why that occurs I don't know. I started digging into the pops code, but quickly got lost.